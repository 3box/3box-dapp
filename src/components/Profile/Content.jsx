import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Route, Switch, withRouter } from 'react-router-dom';

import * as routes from '../../utils/routes';
import StatusUpdate from '../StatusUpdate';
import Activity from './Activity';
import Details from './Details';
import '../styles/Feed.css';
import '../styles/NetworkArray.css';

const ProfileContent = ({ location }) => (
  <div>
    <StatusUpdate />

    {
      location.pathname === routes.PROFILE_ACTIVITY
      && <Activity />
    }

    {
      location.pathname === routes.PROFILE_ABOUT
      && <Details />
    }

    {/* {
      location.pathname === routes.PROFILE_COLLECTIBLES
      && <Collectibles />
    } */}

    {/* <Switch>
      <Route
        exact
        path={routes.PROFILE_ACTIVITY}
        component={() => <Activity />}
      />
      <Route
        exact
        path={routes.PROFILE_ABOUT}
        component={() => <Details />}
      />
      <Route
        exact
        path={routes.PROFILE_COLLECTIBLES}
        component={() => <Collectibles />}
      />
    </Switch> */}

  </div>
);

ProfileContent.propTypes = {
  location: PropTypes.object,
};

ProfileContent.defaultProps = {
  location: {},
};

const mapState = state => ({
  feedByAddress: state.threeBox.feedByAddress,
  ifFetchingActivity: state.threeBox.ifFetchingActivity,
  verifiedGithub: state.threeBox.verifiedGithub,
});

export default withRouter(connect(mapState)(ProfileContent));
