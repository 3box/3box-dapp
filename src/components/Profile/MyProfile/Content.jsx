import React from 'react';
import {
  NavLink, Route, Switch, withRouter,
} from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as routes from '../../../utils/routes';
import StatusUpdate from '../StatusUpdate';
import Activity from './Activity';
import Details from '../Details';
import '../../styles/Feed.css';
import '../../../views/styles/Profile.css';
import '../../styles/NetworkArray.css';

const Content = ({ currentAddress }) => (
  <div>
    <StatusUpdate />
    <div className="profile__category--mobile">
      <div className="profile__category__sectionWrapper">
        <NavLink exact to={`/${currentAddress}/${routes.ACTIVITY}`} className="profile__category__section">Activity</NavLink>
        <NavLink exact to={`/${currentAddress}/${routes.DETAILS}`} className="profile__category__section ">Details</NavLink>
      </div>
    </div>

    <Switch>
      <Route
        exact
        path={routes.FORMAT_PROFILE_ACTIVITY}
        component={Activity}
      />

      <Route
        exact
        path={routes.FORMAT_PROFILE_ABOUT}
        component={Details}
      />
    </Switch>


  </div>
);

Content.propTypes = {
  currentAddress: PropTypes.string,
};

Content.defaultProps = {
  currentAddress: '',
};

const mapState = state => ({
  currentAddress: state.threeBox.currentAddress,
});

export default withRouter(connect(mapState)(Content));
