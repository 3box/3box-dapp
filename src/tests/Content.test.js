import React from 'react';
import ReactDOM from 'react-dom';
import Content from '../components/Content';
import {
  renderWithIntl,
  rendererCreateWithIntl,
} from './helper';

test('renders without crashing', () => {
  const div = document.createElement('div');
  renderWithIntl(<Content />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('render correctly', () => {
  const component = rendererCreateWithIntl(<Content />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
