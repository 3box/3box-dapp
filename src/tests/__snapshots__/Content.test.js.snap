// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<p
  className="App-intro"
>
  To get started, edit  
  <code>
    src/App.js
  </code>
    and save to reload.
</p>
`;
